### PR TITLE
docs(picker): add documentation for the `Searcher` type

### PR DESCRIPTION
**Pull request by adrianschmidt**
## Review:
- [ ] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [ ] Commits have the correct *type* for the changes made
- [ ] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

File: src/components/picker/searcher.types.ts [hunk 1]
```diff
@@ -1,3 +1,11 @@
 import { ListItem } from '../list/list-item.types';
 
+/**
+ * A search function that takes a search-string as an argument, and returns
+ * a promise that will eventually be resolved with an array of `ListItem`:s.
+ *
+ * @param {string} query A search query. Typically what the user has written
+ * in the input field of a limel-picker.
+ * @returns {Promise<ListItem[]>} The search result.
```
Comment (adrianschmidt) [inline src/components/picker/searcher.types.ts:9]: Unfortunately the _types_ disappear. I'm guessing that's a bug, either in lime-elements or in kompendium.

I'm not sure if we should add the types to the description as a workaround, or just try to find and fix the bug instead? @jgroth @mjlime

Comment (mjlime) [reply src/components/picker/searcher.types.ts:9]: I'd love if it was all fixed, but do what you can.

**Review by jgroth — APPROVED** [review]

**Pull request merged**